### PR TITLE
Bug 1241041 - [emulator] Support SSP for qemu bt

### PR DESCRIPTION
**Pull request by shuangMetiz**

### CODE REVIEW

File: hw/bt-hci.c [hunk 1]
```diff
@@ -905,6 +906,21 @@ static int bt_hci_name_req(struct bt_hci_s *hci, bdaddr_t *bdaddr)
...
+    rmt_host_feature_params.features[1]	= (slave->lmp_caps >>  8) & 0xff;
+    rmt_host_feature_params.features[2]	= (slave->lmp_caps >> 16) & 0xff;
+    rmt_host_feature_params.features[3]	= (slave->lmp_caps >> 24) & 0xff;
+    rmt_host_feature_params.features[4]	= (slave->lmp_caps >> 32) & 0xff;
+    rmt_host_feature_params.features[5]	= (slave->lmp_caps >> 40) & 0xff;
+    rmt_host_feature_params.features[6]	= (slave->lmp_caps >> 48) & 0xff;
+    rmt_host_feature_params.features[7]	= (slave->lmp_caps >> 56) & 0xff;
+
+    bt_hci_event(hci, EVT_RMT_HOST_FEATURES_COMPLETE,
+                    &rmt_host_feature_params, EVT_RMT_HOST_FEATURES_COMPLETE_SIZE);
```
Comment (bentian) [inline hw/bt-hci.c:922]: nit: alignment

```
bt_hci_event(hci, EVT_RMT_HOST_FEATURES_COMPLETE,
                      &rmt_host_feature_params, EVT_RMT_HOST_FEATURES_COMPLETE_SIZE);
```

File: hw/bt-hci.c [hunk 2]
```diff
@@ -905,6 +906,21 @@ static int bt_hci_name_req(struct bt_hci_s *hci, bdaddr_t *bdaddr)
...
+    // assume the remote device supports full features
+    slave->lmp_caps = 0xFFFFFFFFFFFFFFFFll;
+    rmt_host_feature_params.features[0]	= (slave->lmp_caps >>  0) & 0xff;
+    rmt_host_feature_params.features[1]	= (slave->lmp_caps >>  8) & 0xff;
+    rmt_host_feature_params.features[2]	= (slave->lmp_caps >> 16) & 0xff;
+    rmt_host_feature_params.features[3]	= (slave->lmp_caps >> 24) & 0xff;
+    rmt_host_feature_params.features[4]	= (slave->lmp_caps >> 32) & 0xff;
+    rmt_host_feature_params.features[5]	= (slave->lmp_caps >> 40) & 0xff;
+    rmt_host_feature_params.features[6]	= (slave->lmp_caps >> 48) & 0xff;
+    rmt_host_feature_params.features[7]	= (slave->lmp_caps >> 56) & 0xff;
```
Comment (bentian) [inline hw/bt-hci.c:919]: Remove extra spaces before =

File: hw/bt-hci.c [hunk 3]
```diff
@@ -1929,6 +1945,11 @@ static void bt_submit_hci(struct HCIInfo *info,
         bt_hci_event_complete_status(hci, HCI_SUCCESS);
         break;
 
+    case cmd_opcode_pack(OGF_HOST_CTL, OCF_WRITE_SIMPLE_PAIRING_MODE):
+        bt_hci_event_complete(hci, &PARAM(write_simple_pairing_mode, simple_pairing_mode),
+            WRITE_SIMPLE_PAIRING_MODE_CP_SIZE);
```
Comment (bentian) [inline hw/bt-hci.c:1950]: nit: alignment

File: hw/bt-hci.c (hunk 2 above)
Comment (shuangMetiz) [inline hw/bt-hci.c:919]: It's a "tab" not empty space, I followed the qemu style.

File: hw/bt-hci.c (hunk 3 above)
Comment (shuangMetiz) [inline hw/bt-hci.c:1950]: OK.

**Pull request merged**